src/tests/unit_tests/test_document_converter.py

```diff
@@ -402,7 +412,64 @@ def test_convert_langchain_document(langchain_doc, firestore_doc):
 )
 def test_roundtrip_firestore(firestore_doc):
     langchain_doc = convert_firestore_document(firestore_doc)
-    roundtrip_doc = convert_langchain_document(langchain_doc, pytest.client)
+    roundtrip_doc = convert_langchain_document(langchain_doc, firestore_client)
 
     assert roundtrip_doc["data"] == firestore_doc.to_dict()
     assert roundtrip_doc["reference"]["path"] == firestore_doc.reference.path
+
+
+@pytest.mark.parametrize(
+    "firestore_doc",
+    [
+        (
+            DocumentSnapshot(
+                reference=DocumentReference(*["foo", "bar"], client=firestore_client),
+                data={
+                    "embedding": Vector([1, 2, 3]),
+                    "content": "test_doc2",
+                },
+                exists=True,
+                read_time=None,
+                create_time=None,
+                update_time=None,
+            )
+        ),
+    ],
+)
+def test_vector_type_from_firestore(firestore_doc):
+    """
+    Test vector type conversion from langchain to firestore and back.
+    """
+    langchain_doc = convert_firestore_document(firestore_doc)
+
+    # Check that the vector fields are removed
+    assert (
+        "embedding" not in langchain_doc.page_content
+        and "embedding" in langchain_doc.metadata
+    )
+    assert "test_doc2" == langchain_doc.page_content
+
+
+def test_vector_type_to_firestore():
```

Comment:
This is doing (A) LangChain --> (B) Firestore  
But is not completing (B) Firestore --> (C) LangChain  and  comparing A == C

The test name indicates is only A-->B   but the test description is says is back to langchain

Reply:
Fixed